### PR TITLE
retrain VAE

### PR DESCRIPTION
**Pull request by ajlee21**
This PR retrains the VAE after discovering that my legends on the loss plot were switched and to make sure the processed data was of the correct dimensions.

Based on the loss cuves (`human_analysis/logs/NN_2500_30/.._hist.svg`) there could still be tuning for this model, especially to increase the number of samples used. Due to local memory constraints I will leave the model as is for now. I also want to make sure I can regenerate previous results that were provided to collaborators using this VAE model but with fewer samples.

Main changes were:
1. Re-running `human_analysis/1_process_recount2_data.ipynb`. Note the change to update the legend can be found in a separate PR (https://github.com/greenelab/ponyo/pull/5) no need for you to review that
2. Creating `explore_data/viz_recount2_compendium.ipynb` to look at the compendium data to get a better sense for why the loss curves look the way they do
3. Update the `generic_expression_patterns_modules/download_recount2_data.R` to take in a parameter to download a user specified number of experiments instead of being fixed.

### CODE REVIEW

File: explore_data/nbconverted/viz_recount2_compendium.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+seed(random_state)
+
+
+# In[2]:
+
+
+# Read in config variables
+base_dir = os.path.abspath(os.path.join(os.getcwd(),"../"))
+
+config_file = os.path.abspath(os.path.join(base_dir,
```
Comment (danich1) [inline explore_data/nbconverted/viz_recount2_compendium.py:56]:
```suggestion
config_file = os.path.abspath(
  os.path.join(
    base_dir,
    "config_human.tsv"
   )
)
```

File: explore_data/nbconverted/viz_recount2_compendium.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+normalized_compendium.head()
+
+
+# In[5]:
+
+
+# Get validation and training set
+# random_state matches the state used in the training
+test_set_percent = validation_frac
+val_df = normalized_compendium.sample(frac=test_set_percent, random_state=123)
```
Comment (danich1) [inline explore_data/nbconverted/viz_recount2_compendium.py:89]:
```suggestion
val_df = (
  normalized_compendium
  .sample(frac=test_set_percent, random_state=123)
)
```

File: explore_data/nbconverted/viz_recount2_compendium.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+# UMAP embedding of original input data
+
+# Get and save model
+model = umap.UMAP(random_state=random_state).fit(normalized_compendium)
+
+input_data_UMAPencoded = model.transform(normalized_compendium)
+input_data_UMAPencoded_df = pd.DataFrame(data=input_data_UMAPencoded,
+                                         index=normalized_compendium.index,
+                                         columns=['1','2'])
```
Comment (danich1) [inline explore_data/nbconverted/viz_recount2_compendium.py:104]:
```suggestion
input_data_UMAPencoded_df = pd.DataFrame(
  data=input_data_UMAPencoded,
  index=normalized_compendium.index,
  columns=['1','2']
)
```

File: human_analysis/logs/NN_2500_30/tybalt_2layer_30latent_stats.tsv [hunk 4]
```diff
@@ -1,41 +1,41 @@
...
-3054.2125204763106	3062.1776258486375	0.001	10	40	0.01
-3051.246826171875	3063.690449085932	0.001	10	40	0.01
-3044.867250504032	3053.3639434180945	0.001	10	40	0.01
-3036.5818737399195	3048.1008814577553	0.001	10	40	0.01
-3046.1041535408267	3044.620108274707	0.001	10	40	0.01
-3044.0268160912296	3040.546544527615	0.001	10	40	0.01
-3033.2976310483873	3044.9473135094177	0.001	10	40	0.01
-3034.587843371976	3032.0526735582944	0.001	10	40	0.01
-3036.5941043976813	3029.345461256263	0.001	10	40	0.01
+2816.8391862709377	3013.397590357973	0.001	10	40	0.01
```
Comment (danich1) [inline human_analysis/logs/NN_2500_30/tybalt_2layer_30latent_stats.tsv:2]: I assume this project was ran using a different dataset. The loss seems to be quite better than the previous run. Is this correct?

File: explore_data/nbconverted/viz_recount2_compendium.py (hunk 2 above)
Comment (ben-heil) [inline explore_data/nbconverted/viz_recount2_compendium.py:89]: The random state here should probably be set with your `random_state` variable

File: explore_data/nbconverted/viz_recount2_compendium.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+print(fig)
+
+
+# **Observations:**
+# * There looks to be a good amount of variance in the compendium overall.
+# * Using a split of 25% seems to get a similar distribution of data between training and validation sets.
+# * Remember, the dataset is in 17K dimensional space, which will make the small clusters difficult to represent during training
+# 
+# Overall, having so many features in our dataset, points to the need for more samples to represent the structure in the compendium. For now, we are limited by memory to only select a subset of recount2, but in a future iteration perhaps this will be updated.
```
Comment (ben-heil) [inline explore_data/nbconverted/viz_recount2_compendium.py:143]: In what way are you memory limited? If you're training a VAE, couldn't you use a [Tensorflow Dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset) so that you don't have to fit all the data into memory during training?

File: human_analysis/logs/NN_2500_30/tybalt_2layer_30latent_stats.tsv (hunk 4 above)
Comment (ajlee21) [reply human_analysis/logs/NN_2500_30/tybalt_2layer_30latent_stats.tsv:2]: The dataset I used originally, that was part of the grant, was a smaller dataset of ~1.5K samples and 10% were held out for validation. This new dataset is ~5K samples with 25% held out for validation. So the new loss is lower, though there is still a fair gap between the validation and training loss that hopefully can be reduced with more tuning in the future

File: explore_data/nbconverted/viz_recount2_compendium.py (hunk 5 above)
Comment (ajlee21) [reply explore_data/nbconverted/viz_recount2_compendium.py:153]: Currently, I am running this on my office desktop and the full dataset used for training is being held in memory so I am limited by the memory on this machine. But I have talked with Casey about training on the full recount2 dataset by reading the data into batches as youre suggested. The downloading of the data from recount2 may also be contributing to the memory issue but I have not fully tested this yet. I will make an issue for the future to address this.
